src/JBDistributor.sol

```diff
@@ -1,208 +1,286 @@
-     */
-    function collectVestedRewards(uint256[] calldata _tokenIds, IERC20[] calldata _tokens, uint256 _round) public {
-        // Make sure the vesting is done
-        if (_round > currentRound()) {
-            revert NotVestedYet();
+    /// @notice Allows an address to collect vested tokens for various rounds.
+    /// @param _rounds The rounds to collect for.
+    function collectVestedRewards(CollectVestingRoundData[] calldata _rounds) external {
+        // TODO: We can optimize this call by batching transfers
+        // Collect for each specified round.
+        for (uint256 _i = 0; _i < _rounds.length;) {
+            collectVestedRewards(_rounds[_i].tokenIds, _rounds[_i].tokens, _rounds[_i].round, _rounds[_i].beneficiary);
+
+            unchecked {
+                ++_i;
+            }
         }
+    }
+    
+    //*********************************************************************//
+    // ----------------------- public transactions ----------------------- //
+    //*********************************************************************// 
 
-        for (uint256 _i; _i < _tokens.length;) {
+    /// @notice Collect vested tokens.
+    /// @param _tokenIds The IDs of the 721s to claim for.
+    /// @param _tokens The address of the tokens being claimed.
+    /// @param _round The round during which the tokens were done vesting.
+    function collectVestedRewards(uint256[] calldata _tokenIds, IERC20[] calldata _tokens, uint256 _round, address _beneficiary) public {
+        // Make sure the specified round has passed.
+        if (_round > currentRound()) revert NotVestedYet();
+
+        // Keep a reference to the number of tokens.
+        uint256 _numberOfTokens = _tokens.legnth; 
+
+        // Keep a reference to the token being iterated on.
+        IERC20 _token;
+
+        // Loop through each token of which vested rewards are being collected.
+        for (uint256 _i; _i < _numberOfTokens;) {
+
+            // Set the token.
+            _token = _tokens[_i];
+
+            // Keep a reference to the total amount of tokens there are.
             uint256 _totalTokenAmount;
 
-            for (uint256 _j; _j < _tokenIds.length;) {
+            // Keep a reference to the number of token IDs.
+            uint256 _numberOfTokenIds = _tokenIds.legnth; 
+
+            // Keep a reference to the ID of the token being iterated on.
+            uint256 _tokenId;
+
+            /// Loop through each token ID for which vested rewards are being collected.
+            for (uint256 _j; _j < _numberOfTokenIds;) {
+                // Set the token ID.
+                _tokenId = _tokenIds[_j];
+
                 // TODO: make sure the sender owns the tokenId, this also makes sure it is not burned
-                if (_i == 0 && !_canClaim(_tokenIds[_j], msg.sender)) revert NoAccess();
+                if (_i == 0 && !_canClaim(_tokenId, msg.sender)) revert NoAccess();
```

Comment:
if i != 0, we don't check if can claim?

Reply:
j is the NFT ownership, we do indeed wanna check each of those at least once (i == 0).
i is the token being claimed. once it is not 0, we can be sure that the ownership of j has already been checked.
(pretty sure, but may be off.)

maybe the loop order makes this feel strange? should i loop over NFTs and j loop over tokens being claimed?

Reply:
> j is the NFT ownership, we do indeed wanna check each of those at least once (i == 0).
i is the token being claimed. once it is not 0, we can be sure that the ownership of j has already been checked.

You got it, this order was done to make batch transfers easier, if we change the loop order we have to track all the amounts we want to send out and do another loop at the function to do the transfers. 

We can also do a loop at the top of this function to perform these checks, but I liked the simplicity of doing it this way.

Reply:
oh yes, got it, was mentally inverting the 2 loops (can claim if nft is ok, not a per token basis), all good